plugins/inventory/now.py

```diff
@@ -0,0 +1,443 @@
+                return True
+            # TODO remove?
+            self.display.vvv(
+                'Skipping due to inventory source not ending in "now.yaml" nor "now.yml"'
+            )
+            return False
+
+    def validate_grouping_conditions(self, parent):
+        for category, val in parent.items():
+            if val and all(i in val for i in ["includes", "excludes"]):
+                raise AnsibleParserError(
+                    "Invalid configuration: 'includes' and 'excludes' are mutually exclusive."
+                )
+
+    def _includes_query(self, column, includes):
+        return "^OR".join("{0}={1}".format(column, i) for i in includes)
+
+    def _excludes_query(self, column, excludes):
+        return "^".join("{0}!={1}".format(column, i) for i in excludes)
+
+    def _sysparm_query(self, conditions):
+        param_queries = []
+        for column, val in conditions.items():
+            if val:
+                includes = val.get("includes")
+                if includes:
+                    param_queries.append(self._includes_query(column, includes))
+                excludes = val.get("excludes")
+                if excludes:
+                    param_queries.append(self._excludes_query(column, excludes))
+        if param_queries:
+            return "^".join(i for i in param_queries)
+        return None
+
+    def _query(self, conditions):
+        columns = self.get_option("columns")
+        ansible_host_source = self.get_option("ansible_host_source")
+        inventory_hostname_source = self.get_option("inventory_hostname_source")
+
+        requested_fields = set(
+            columns
+            + ["sys_id", ansible_host_source, inventory_hostname_source]
+            + list(conditions.keys())
+        )
+        query = dict(
+            # Request only the table columns we're interested in
+            sysparm_fields=",".join(requested_fields),
+            # Make references and choice fields human-readable
+            sysparm_display_value=True,
+        )
+        sysparm_query = self._sysparm_query(conditions)
+        if sysparm_query:
+            query["sysparm_query"] = sysparm_query
+
+        return query
+
+    def add_host(self, record):
+        ansible_host_source = self.get_option("ansible_host_source")
+        inventory_hostname_source = self.get_option("inventory_hostname_source")
+
+        if inventory_hostname_source not in record:
+            # The user specified an invalid column
+            raise AnsibleParserError(
+                "Inventory hostname source column '{0}'"
+                " is not present in table {1}.".format(
+                    inventory_hostname_source, self.get_option("table")
+                )
+            )
+
+        inventory_hostname = record[inventory_hostname_source]
+        if inventory_hostname:
+            host = self.inventory.add_host(inventory_hostname)
+            # TODO: what if the value is empty, do we handle this?
```

Comment:
I would say we print the warning and skip the bad entry. In the future, we can then add a configuration option that will toggle the behavior between skip and abort (if anyone will be interested in this kind of functionality).